### PR TITLE
Refactoring of boundary classes

### PR DESCRIPTION
**Pull request by LinhWuerzburger**
refactoring to switch to an underscore notation. only changing the files of the boundary folder

### CODE REVIEW

File: src/advection/SLAdvect.cpp [hunk 1]
```diff
@@ -98,48 +98,48 @@ void SLAdvect::advect(Field *out, Field *in, const Field *u_vel, const Field *v_
...
-            if (Ci > 0) {
+            if (Ci > epsilon_x) {
                 i0 = std::max(i_start, (i - static_cast<long int>(Ci)));
                 i1 = i0 - 1;
-                r = fabs(fmod(Ci, 1));
-            } else {
-                i1 = std::min(i_end, i - static_cast<long int>(Ci));
-                i0 = i1 + 1;
-                r = 1 - fabs(fmod(Ci, 1));
+                r = fmod(Ci, 1); // closer to i1 if r is closer to 1
```
Comment (ChristianVonMach) [inline src/advection/SLAdvect.cpp:108]: needs two spaces between code and comment

File: src/advection/SLAdvect.cpp [hunk 2]
```diff
@@ -98,48 +98,48 @@ void SLAdvect::advect(Field *out, Field *in, const Field *u_vel, const Field *v_
...
+            } else if (Ci < -epsilon_x) {
+                i0 = std::min(i_end, i - static_cast<long int>(Ci));
+                i1 = i0 + 1;
+                r = fmod(-Ci, 1); // closer to i1 if r is closer to 1
```
Comment (ChristianVonMach) [inline src/advection/SLAdvect.cpp:112]: needs two spaces between code and comment

File: src/boundary/Boundary.h [hunk 3]
```diff
@@ -15,64 +15,65 @@
...
+    size_t* get_inner_list() { return m_inner_list;}
+    size_t get_size_inner_list() { return m_size_innerList; }
+
+    size_t get_size_boundary_list() {return m_size_boundary_list;}
+    size_t get_size_boundary_front() {return m_size_boundary_front;}
+    size_t get_size_boundary_back() {return m_size_boundary_back;}
+    size_t get_size_boundary_top() {return m_size_boundary_top;}
+    size_t get_size_boundary_bottom() {return m_size_boundary_bottom;}
+    size_t get_size_boundary_left() {return m_size_boundary_left;}
+    size_t get_size_boundary_right() { return m_size_boundary_right;}
```
Comment (ChristianVonMach) [inline src/boundary/Boundary.h:39]: those can be marked as const

File: src/boundary/BoundaryController.cpp [hunk 4]
```diff
@@ -5,146 +5,172 @@
...
     //}
-    //delete[] m_surfaceList;
-    //for (size_t obstacle = 0; obstacle < m_numberOfObstacles; obstacle++) {
-    //    delete (*(m_obstacleList + obstacle));
+    //delete[] m_surface_list;
+    //for (size_t obstacle = 0; obstacle < m_number_of_obstacles; obstacle++) {
+    //    delete (*(m_obstacle_list + obstacle));
     //}
-    //delete[] m_obstacleList;
+    //delete[] m_obstacle_list;
```
Comment (ChristianVonMach) [inline src/boundary/BoundaryController.cpp:173]: dead code

File: src/boundary/BoundaryController.h [hunk 5]
```diff
@@ -7,52 +7,52 @@
...
+    size_t get_size_surfaceList() {return m_size_surface_list;};
 
-    size_t getObstacleStrideX(size_t id, size_t level);
-    size_t getObstacleStrideY(size_t id, size_t level);
-    size_t getObstacleStrideZ(size_t id, size_t level);
+    size_t get_obstacle_stride_x(size_t id, size_t level);
+    size_t get_obstacle_stride_y(size_t id, size_t level);
+    size_t get_obstacle_stride_z(size_t id, size_t level);
 
     std::vector<FieldType> get_used_fields();
```
Comment (ChristianVonMach) [inline src/boundary/BoundaryController.h:53]: getters can be marked as const

File: src/boundary/BoundaryData.h [hunk 6]
```diff
@@ -33,32 +33,33 @@ enum BoundaryCondition : int {
...
+    static FieldType match_field(const std::string& string);
+    static Patch match_patch(const std::string& string);
+    static BoundaryCondition match_boundary_condition(const std::string& string);
 
-    void addBoundaryCondition(const std::vector<Patch>& patches, real value, BoundaryCondition boundaryCondition);
-    BoundaryCondition getBoundaryCondition(Patch p){ return m_boundaryConditions[p];};
-    real getValue(Patch p){ return m_values[p];};
+    void add_boundary_condition(const std::vector<Patch>& patches, real value, BoundaryCondition boundary_condition);
+    BoundaryCondition get_boundary_condition(Patch p) { return m_boundary_conditions[p];}
+    real get_value(Patch p) { return m_values[p];}
```
Comment (ChristianVonMach) [inline src/boundary/BoundaryData.h:51]: can be const

File: src/boundary/BoundaryData.h [hunk 7]
```diff
@@ -33,32 +33,33 @@ enum BoundaryCondition : int {
...
     BoundaryData();
     ~BoundaryData();
     void print();
 
-    static std::string getFieldTypeName(FieldType f);
-    static std::string getBoundaryConditionName(BoundaryCondition bc);
-    static std::string getPatchName(Patch p);
+    static std::string get_field_type_name(FieldType f);
+    static std::string get_boundary_condition_name(BoundaryCondition bc);
+    static std::string get_patch_name(Patch p);
```
Comment (ChristianVonMach) [inline src/boundary/BoundaryData.h:43]: can be const

File: src/boundary/Multigrid.cpp [hunk 8]
```diff
@@ -31,70 +33,70 @@ Multigrid::Multigrid(BoundaryDataController *bdc_boundary) {
...
-        m_MG_oLeft = new size_t *[m_levels + 1];
-        m_MG_oRight = new size_t *[m_levels + 1];
-
-        //start index of each obstacle in level joined list (SliceZ = Front/Back, SliceY = Bottom/Top, SliceX = Left/Right)
+        // m_MG_oFront = new size_t *[m_levels + 1];
+        // m_MG_oBack = new size_t *[m_levels + 1];
+        // m_MG_oBottom = new size_t *[m_levels + 1];
+        // m_MG_oTop = new size_t *[m_levels + 1];
+        // m_MG_oLeft = new size_t *[m_levels + 1];
+        // m_MG_oRight = new size_t *[m_levels + 1];
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.cpp:73]: dead code

File: src/boundary/Multigrid.cpp [hunk 9]
```diff
@@ -103,10 +105,10 @@ Multigrid::Multigrid(size_t numberOfSurfaces, Surface **surfaceList, size_t numb
     addMGLists();
     sendListsToGPU();
 #ifndef BENCHMARKING
-    //print();
+    // print();
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.cpp:108]: dead code

File: src/boundary/Multigrid.cpp [hunk 10]
```diff
@@ -171,13 +171,13 @@ Multigrid::~Multigrid() {
...
-            delete (*(m_MG_oTop + level));
-            delete (*(m_MG_oLeft + level));
-            delete (*(m_MG_oRight + level));
+        //    delete (*(m_MG_oList + level));
+        //    delete (*(m_MG_oFront + level));
+        //    delete (*(m_MG_oBack + level));
+        //    delete (*(m_MG_oBottom + level));
+        //    delete (*(m_MG_oTop + level));
+        //    delete (*(m_MG_oLeft + level));
+        //    delete (*(m_MG_oRight + level));
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.cpp:180]: dead code

File: src/boundary/Multigrid.cpp [hunk 11]
```diff
@@ -224,25 +224,25 @@ Multigrid::~Multigrid() {
...
+//        size_t size_oBottom = getLen_oBottom_joined();
+//        size_t size_oTop = getLen_oTop_joined();
+//        size_t size_oLeft = getLen_oLeft_joined();
+//        size_t size_oRight = getLen_oRight_joined();
+// #pragma acc exit data delete(m_data_MG_oFront_level_joined[:size_oFront])
+// #pragma acc exit data delete(m_data_MG_oBack_level_joined[:size_oBack])
+// #pragma acc exit data delete(m_data_MG_oTop_level_joined[:size_oTop])
+// #pragma acc exit data delete(m_data_MG_oBottom_level_joined[:size_oBottom])
+// #pragma acc exit data delete(m_data_MG_oLeft_level_joined[:size_oLeft])
+// #pragma acc exit data delete(m_data_MG_oRight_level_joined[:size_oRight])
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.cpp:245]: dead code

File: src/boundary/Multigrid.h [hunk 12]
```diff
@@ -44,36 +44,30 @@ class Multigrid {
     size_t getObstacleStrideY(size_t id, size_t level);
     size_t getObstacleStrideZ(size_t id, size_t level);
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.h:45]: could also be renamed

File: src/boundary/Multigrid.h [hunk 13]
```diff
@@ -8,31 +8,31 @@
...
+    size_t getSize_innerList_level_joined() { return *(m_size_MG_iList_level + m_levels + 1); }
     size_t getInnerList_level_joined_start(size_t level);
     size_t getInnerList_level_joined_end(size_t level);
 
-    size_t* getBoundaryList_level_joined() { return m_data_MG_bList_level_joined; };
-    size_t getSize_boundaryList_level_joined() { return *(m_size_MG_bList_level + m_levels + 1); };
+    size_t* getBoundaryList_level_joined() { return m_data_MG_bList_level_joined; }
+    size_t getSize_boundaryList_level_joined() { return *(m_size_MG_bList_level + m_levels + 1); }
     size_t getBoundaryList_level_joined_start(size_t level);
     size_t getBoundaryList_level_joined_end(size_t level);
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.h:37]: can be const

File: src/boundary/Multigrid.h [hunk 14]
```diff
@@ -178,13 +168,15 @@ class Multigrid {
     void print();
 
     size_t** m_data_boundary_patches_joined;
-    //size_t** m_data_surfaces_patches_joined;
+    // size_t** m_data_surfaces_patches_joined;
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.h:171]: dead code

File: src/boundary/Multigrid.h [hunk 15]
```diff
@@ -178,13 +168,15 @@ class Multigrid {
...
     size_t** m_data_obstacles_patches_joined;
     BoundaryDataController* m_bdc_boundary;
     BoundaryDataController** m_bdc_obstacle;
 
     void removeBoundaryListsFromGPU();
-};
 
+    void controlObstacleOverlap(Obstacle* o, size_t *i1, size_t *i2, size_t *j1, size_t *j2, size_t *k1, size_t *k2);
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.h:178]: should use _

File: src/boundary/Multigrid.h [hunk 16]
```diff
@@ -145,12 +135,12 @@ class Multigrid {
...
-    size_t getLastIndex_bSliceY( size_t level);
+    size_t getFirstIndex_bSliceZ(size_t level);
+    size_t getFirstIndex_bSliceX(size_t level);
+    size_t getFirstIndex_bSliceY(size_t level);
+    size_t getLastIndex_bSliceZ(size_t level);
+    size_t getLastIndex_bSliceX(size_t level);
+    size_t getLastIndex_bSliceY(size_t level);
 
     size_t getLen_iList_joined();
     size_t getLen_bList_joined();
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.h:146]: can be marked as const

File: src/boundary/Obstacle.cpp [hunk 17]
```diff
@@ -106,8 +90,8 @@ void Obstacle::init(size_t level) {
     m_obstacleLeft = new size_t[m_size_obstacleLeft];
     m_obstacleRight = new size_t[m_size_obstacleRight];
 
-    //m_size_obstacleInner = (m_strideX - 2) * (m_strideY - 2) * (m_strideZ - 2);
-    //m_obstacleInner = new size_t[m_size_obstacleInner];
+    // m_size_obstacleInner = (m_strideX - 2) * (m_strideY - 2) * (m_strideZ - 2);
+    // m_obstacleInner = new size_t[m_size_obstacleInner];
     createObstacle(Nx, Ny);
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.cpp:95]: dead code

Comment (ChristianVonMach) [inline src/boundary/Obstacle.cpp:95]: could be renamed with _

File: src/boundary/Obstacle.cpp [hunk 18]
```diff
@@ -122,7 +106,7 @@ Obstacle::~Obstacle() {
     delete (m_obstacleBottom);
     delete (m_obstacleLeft);
     delete (m_obstacleRight);
-    //delete (m_obstacleInner);
+    // delete (m_obstacleInner);
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.cpp:109]: dead code

File: src/boundary/Obstacle.cpp [hunk 19]
```diff
@@ -210,18 +194,18 @@ void Obstacle::createObstacle(size_t Nx, size_t Ny) {
...
-    //            size_t idx = IX(i, j, k, strideX, strideY);
-    //            *(m_obstacleInner + index) = m_obstacleList[idx];
-    //        }
-    //    }
-    //}
+    //             size_t idx = IX(i, j, k, strideX, strideY);
+    //             *(m_obstacleInner + index) = m_obstacle_list[idx];
+    //         }
+    //     }
+    // }
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.cpp:208]: dead code

File: src/boundary/Obstacle.h [hunk 20]
```diff
@@ -39,7 +40,7 @@ class Obstacle {
     size_t getSize_obstacleLeft() { return m_size_obstacleLeft; }
     size_t getSize_obstacleRight() { return m_size_obstacleRight; }
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.h:41]: can be marked const

File: src/boundary/Obstacle.h [hunk 21]
```diff
@@ -39,7 +40,7 @@ class Obstacle {
...
 
-    bool isObstacleCell(size_t i, size_t j, size_t k);
+    bool isObstacleCell(size_t i, size_t j, size_t k) const;
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.h:43]: could be renamed with _

File: src/boundary/Obstacle.h [hunk 22]
```diff
@@ -54,28 +55,42 @@ class Obstacle {
     size_t getStrideY() { return m_j2 - m_j1 + 1; }
     size_t getStrideZ() { return m_k2 - m_k1 + 1; }
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.h:56]: could be renamed with _ and be marked const

File: src/boundary/Obstacle.h [hunk 23]
```diff
@@ -85,13 +100,26 @@ class Obstacle {
 
     void init(size_t level);
     void createObstacle(size_t Nx, size_t Ny);
-    void control();
 
-    static real matchGrid(double obstacleCoordinate, real dx, real x1);
+    static real match_grid(real obstacle_coordinate, real spacing, real start_coordinate);
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.h:104]: could be renamed with _

File: src/utility/Utility.cpp [hunk 24]
```diff
@@ -114,23 +80,134 @@ std::shared_ptr<spdlog::logger> create_logger(std::string logger_name) {
...
+    }
+    // size_t Nx = Domain::getInstance()->get_Nx();
+    // size_t Ny = Domain::getInstance()->get_Ny();
+    // std::string index;
+    // for (size_t idx: indices) {
+    //     size_t k = getCoordinateK(idx, Nx, Ny);
+    //     size_t j = getCoordinateJ(idx, Nx, Ny, k);
+    //     size_t i = getCoordinateI(idx, Nx, Ny, j, k);
+    //     index += " (" + std::to_string(i) + "|" + std::to_string(j) + "|" + std::to_string(k) + ")";
+    // }
```
Comment (ChristianVonMach) [inline src/utility/Utility.cpp:188]: dead code

**Review by ChristianVonMach — CHANGES_REQUESTED** [review]

**Comment by LinhWuerzburger** [comment]
refactoring done

File: src/advection/SLAdvect.cpp (hunk 1 above)
Comment (LinhWuerzburger) [reply src/advection/SLAdvect.cpp]: not part of this PR anymore

File: src/advection/SLAdvect.cpp (hunk 2 above)
Comment (LinhWuerzburger) [reply src/advection/SLAdvect.cpp]: not part of this PR anymore

File: src/boundary/Boundary.h (hunk 3 above)
Comment (LinhWuerzburger) [reply src/boundary/Boundary.h]: done in 7589a822a5aab780bc1222f574fa636195f8c9c4 and 48c839cb6896a4317350bea2e7f1fe6525a4a4ec

File: src/boundary/BoundaryController.cpp (hunk 4 above)
Comment (LinhWuerzburger) [reply src/boundary/BoundaryController.cpp]: 08117070ac88bab6dedcd4130c7cf13e07891575

File: src/boundary/Obstacle.h (hunk 21 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.h]: done... somewhere

File: src/boundary/Obstacle.h (hunk 22 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.h]: done somewhere

File: src/boundary/Obstacle.h (hunk 23 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.h:101]: done somewhere

File: src/utility/Utility.cpp (hunk 24 above)
Comment (LinhWuerzburger) [reply src/utility/Utility.cpp]: not part of the PR anymore. but I think it was already changed

File: src/boundary/Obstacle.h (hunk 20 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.h]: done somewhere

File: src/boundary/BoundaryData.h (hunk 6 above)
Comment (LinhWuerzburger) [reply src/boundary/BoundaryData.h]: ded5055d03973e886d37103a145ecc5db868bd9f

File: src/boundary/Multigrid.cpp (hunk 8 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp]: done somewhere

File: src/boundary/Multigrid.cpp (hunk 9 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp]: acc6732050bd2767e42f59c0ac68a07c35b15ada

File: src/boundary/Multigrid.cpp (hunk 10 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp]: done somewhere

File: src/boundary/Multigrid.cpp (hunk 11 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp]: done somewhere

File: src/boundary/Multigrid.h (hunk 12 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.h]: done somewhere

File: src/boundary/Multigrid.h (hunk 15 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.h]: done somewhere

File: src/boundary/Obstacle.cpp (hunk 17 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.cpp]: done somewhere

Comment (LinhWuerzburger) [reply src/boundary/Obstacle.cpp]: c485ae6c74c19ffe335f50f77aef3ee74c5c84f2

File: src/boundary/Obstacle.cpp (hunk 18 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.cpp]: done somewhere

File: src/boundary/Obstacle.cpp (hunk 19 above)
Comment (LinhWuerzburger) [reply src/boundary/Obstacle.cpp]: done somewhere

File: src/boundary/BoundaryController.h (hunk 5 above)
Comment (LinhWuerzburger) [reply src/boundary/BoundaryController.h]: d56e0d5ac0a413b529789239ea126a66b109026f

File: src/boundary/BoundaryData.h (hunk 7 above)
Comment (LinhWuerzburger) [reply src/boundary/BoundaryData.h:43]: static member cannot have a const qualifier

File: src/boundary/Multigrid.h (hunk 13 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.h]: c984e0880c80fa8fa1efe7d96bd03afc2407374c

File: src/boundary/Multigrid.h (hunk 16 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.h]: c984e0880c80fa8fa1efe7d96bd03afc2407374c

File: src/boundary/Obstacle.h [hunk 25]
```diff
@@ -20,40 +20,41 @@
...
-    size_t* getObstacleBottom() { return m_obstacleBottom; }
-    size_t* getObstacleLeft() { return m_obstacleLeft; }
-    size_t* getObstacleRight() { return m_obstacleRight; }
+    size_t* get_obstacle_list() { return m_obstacle_list; }
+    size_t* get_obstacle_front() { return m_obstacle_front; }
+    size_t* get_obstacle_back() { return m_obstacle_back; }
+    size_t* get_obstacle_top() { return m_obstacle_top; }
+    size_t* get_obstacle_bottom() { return m_obstacle_bottom; }
+    size_t* get_obstacle_left() { return m_obstacle_left; }
+    size_t* get_obstacle_right() { return m_obstacle_right; }
```
Comment (ChristianVonMach) [inline src/boundary/Obstacle.h:33]: could be const

Comment (LinhWuerzburger) [reply src/boundary/Obstacle.h]: 85bf1063a3b2f3a9b442eb00a8da84c334d4611a

File: src/boundary/Multigrid.cpp (hunk 9 above)
Comment (ChristianVonMach) [reply src/boundary/Multigrid.cpp]: dead code?

**Review by ChristianVonMach — APPROVED** [review]

File: src/boundary/Boundary.h [hunk 26]
```diff
@@ -15,31 +15,31 @@
...
-    void updateLists();
+    size_t* get_boundary_list() const {return m_boundary_list;}
+    size_t* get_boundary_front() const {return m_boundary_front;}
+    size_t* get_boundary_back() const {return m_boundary_back;}
+    size_t* get_boundary_top() const {return m_boundary_top;}
+    size_t* get_boundary_bottom() const {return m_boundary_bottom;}
+    size_t* get_boundary_left() const {return m_boundary_left;}
+    size_t* get_boundary_right() const { return m_boundary_right;}
+
+    size_t* get_inner_list() { return m_inner_list;}
```
Comment (ChristianVonMach) [inline src/boundary/Boundary.h:30]: can be const

Comment (LinhWuerzburger) [reply src/boundary/Boundary.h]: 71d6ea6acf5bf89eeebaa98ea5d1878cfbd1695d

File: src/boundary/Multigrid.cpp [hunk 27]
```diff
@@ -697,115 +719,147 @@ Obstacle** Multigrid::obstacleDominantRestriction(size_t level) {
...
             m_logger->warn("Be cautious! Obstacle fills up inner cells in y-direction at level {}", level);
         }
-        if (k2_fine - k1_fine +1< domain->get_nz(level - 1) - 2 && k2_coarse - k1_coarse + 1 >= domain->get_nz(level) - 2) {
+        if (k2_fine - k1_fine + 1 < domain->get_nz(level - 1) - 2
+            && k2_coarse - k1_coarse + 1 >= domain->get_nz(level) - 2) {
             m_logger->warn("Be cautious! Obstacle fills up inner cells in z-direction at level {}", level);
         }
+
+        for (size_t c = 0; c < id; c++) {
+            bool overlap = control_obstacle_overlap(obstacleList_coarse[c], &i1_coarse, &i2_coarse,
```
Comment (ChristianVonMach) [inline src/boundary/Multigrid.cpp:737]: this var seem unused?

**Review by ChristianVonMach — CHANGES_REQUESTED** [review]

File: src/boundary/Multigrid.cpp (hunk 27 above)
Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp:737]: which var? c? it's used. `obstacleList_coarse[c]` though I see, that I forgot to rename the obstacleList to an underscore notation.

Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp:737]: ah the bool var. yeah, not intentional. just a fail.

Comment (LinhWuerzburger) [reply src/boundary/Multigrid.cpp]: done in 51a193aef05a7a9850e261de0b6ef6641c2831b4

**Review by ChristianVonMach — APPROVED** [review]

**Pull request merged**